### PR TITLE
SPARKC-570 Add ReadConf parameter to Cassandra table join methods

### PR DESCRIPTION
**Pull request by mathgeek**

### CODE REVIEW

**Comment by mathgeek** [comment]
https://datastax-oss.atlassian.net/projects/SPARKC/issues/SPARKC-570

File: spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/RDDFunctions.scala [hunk 1]
```diff
@@ -152,7 +152,8 @@ class RDDFunctions[T](rdd: RDD[T]) extends WritableToCassandra[T] with Serializa
   def joinWithCassandraTable[R](
     keyspaceName: String, tableName: String,
     selectedColumns: ColumnSelector = AllColumns,
-    joinColumns: ColumnSelector = PartitionKeyColumns)(
+    joinColumns: ColumnSelector = PartitionKeyColumns,
+    readConf: ReadConf = ReadConf.fromSparkConf(rdd.sparkContext.getConf))(
```
Comment (edeesis) [inline spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/RDDFunctions.scala]: should these be implicits like SparkContextFunctions#cassandraTable?

Comment (mathgeek) [reply spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/RDDFunctions.scala]: I tried to keep this consistent with other functions in this file, which pass a defaulted WriteConf (RDDFunctions#saveToCassandra)

**Comment by RussellSpitzer** [comment]
I think we are also gonna need at least one integration test making sure we respect the readconf passed in, but other than that this seems very reasonable to me.

**Comment by mathgeek** [comment]
I found out that I apparently need to double check with my company's legal department as well before contributing 😅 Once I get that approval I'll add a test. Thanks for looking!

**Comment by RussellSpitzer** [comment]
Ha, well I can always do a black box implementation of this as well :) I promise I won't look too close

**Comment by RussellSpitzer** [comment]
Since I just clobbered master this will need a slight rebase (just needs to be relocated to the new module location)

**Comment by RussellSpitzer** [comment]
https://github.com/datastax/spark-cassandra-connector/pull/1242 Working on rebase here

**Pull request closed without merging**